shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/RouteCardDataLeafTest.kt

```diff
@@ -387,9 +408,815 @@ class RouteCardDataLeafTest {
+        )
+    }
+
+    @Test
+    fun `formats Red Line southbound as branching even if next trips all match`() = parametricTest {
+        val objects = RedLine.objects()
+        val now = Clock.System.now()
+        val prediction1 =
+            objects.prediction {
+                arrivalTime = now + 2.minutes
+                departureTime = arrivalTime
+                trip = objects.trip(RedLine.ashmontSouth)
+                stopId = RedLine.jfkUmass.south1.id
+            }
+        val prediction2 =
+            objects.prediction {
+                arrivalTime = now + 5.minutes
+                departureTime = arrivalTime
+                trip = objects.trip(RedLine.ashmontSouth)
+                stopId = RedLine.jfkUmass.south1.id
+            }
+        val prediction3 =
+            objects.prediction {
+                arrivalTime = now + 9.minutes
+                departureTime = arrivalTime
+                trip = objects.trip(RedLine.ashmontSouth)
+                stopId = RedLine.jfkUmass.south1.id
+            }
+
+        assertEquals(
+            LeafFormat.branched {
+                branch("Ashmont", UpcomingFormat.Some(
+                    UpcomingFormat.Some.FormattedTrip(
+                        objects.upcomingTrip(prediction1),
+                        RouteType.HEAVY_RAIL,
+                        TripInstantDisplay.Minutes(2)
+                    ),
+                    null
+                ))
+                branch("Ashmont", UpcomingFormat.Some(
+                    UpcomingFormat.Some.FormattedTrip(
+                        objects.upcomingTrip(prediction2),
+                        RouteType.HEAVY_RAIL,
+                        TripInstantDisplay.Minutes(5)
+                    ),
+                    null
+                ))
+                branch("Ashmont", UpcomingFormat.Some(
+                    UpcomingFormat.Some.FormattedTrip(
+                        objects.upcomingTrip(prediction3),
+                        RouteType.HEAVY_RAIL,
+                        TripInstantDisplay.Minutes(9)
+                    ),
+                    null
+                ))
+            },
+            RouteCardData.Leaf(
+                0,
+                listOf(RedLine.ashmontSouth, RedLine.braintreeSouth),
+                setOf(RedLine.jfkUmass.south1.id, RedLine.jfkUmass.south2.id),
+                listOf(objects.upcomingTrip(prediction1), objects.upcomingTrip(prediction2), objects.upcomingTrip(prediction3)),
+                emptyList(),
+                allDataLoaded = true,
+                hasSchedulesToday = true
+            )
+                .format(now, RedLine.route, RedLine.global, anyEnumValue())
+        )
+    }
+
+    @Test
+    @Ignore // TODO once alerts are added
+    fun `formats Red Line southbound as non-branching if one branch is closed`() = parametricTest {
+        TODO("⚠ Southbound to Ashmont 3 min 12 min")
+    }
```

Comment:
There are a handful of these that I’m leaving as placeholders until [Group by direction | Add in alerts data](https://app.asana.com/0/1205732265579288/1209536961544708/f) is ready. I could be trying to define some test data now, but it’s hard to be confident that it’d be the right test data until the functionality actually works.